### DISCUSSION TITLE
compile esp-thread-br, Too many open files

### DISCUSSION THREAD

**Discussion by mvcorrea-ufc** [containers/podman#21100] [General]
situation:

> podman run --rm -v $PWD/project:/project -w /project espressif/idf:latest ./run.sh

On the shell script..

> #!/bin/bash
> 
> S3='/project/esp-thread-br/examples/basic_thread_border_router'`
> if [ -d '/project/esp-thread-br/' ]; then
> rm -rf '/project/esp-thread-br/'
> fi
> 
> ulimit -n 4096 &&
> git clone --recursive https://github.com/espressif/esp-thread-br.git &&
> cp ./sdkconfig $S3/sdkconfig.defaults &&
> 
> #build the H2 firmware
> echo "---> building the esp32H3 firmware -------------------------------------------------------" &&
> cd ${IDF_PATH}/examples/openthread/ot_rcp &&
> idf.py set-target esp32h2 &&
> idf.py build &&
> 
> #build the S3 firmware
> echo "---> building the esp32S3 firmware -------------------------------------------------------" &&
> cd $S3 &&
> idf.py set-target esp32s3 &&
> idf.py build
> 

changes on the sdkconfig:

> ESP Thread Border Router Example --->
> [\*] Enable the automatic start mode in Thread Border Router.
> [\*] Enable the web server in Thread Border Router.
> Example Connection Configuration --->
> (mySSID) WiFi SSID
> (myPASS) WiFi Password

Desperate here! already tried to do it by hand one command after other, changed the ulimit on the linux container, and always the same error "too many open files" when finishing the S3 part, any help is welcome :)
following: https://openthread.io/guides/border-router/espressif-esp32 same hardware.

also put this thread at [openthread](https://github.com/orgs/openthread/discussions/9746)

**Comment by rhatdan** [comment]
Have you tried to run this in rootful mode? 

You might need to change the ulimit on the user account, if you are running in rootless mode.  Podman can not increase the ulimit for a rootless user above the ulimit max set for the user.
